rust.ungram

```diff
@@ -603,7 +603,12 @@ WildcardPat =
   '_'
 
 RangePat =
-  start:Pat op:('..' | '..=') end:Pat
+  // 1..
+  start:Pat op:('..' | '..=')
+  // 1..2
+  | start:Pat op:('..' | '..=') end:Pat
+  // ..2
+  | op:('..' | '..=') end:Pat
```

Comment:
Let's fold this into the same case, like we did for RangeExpr

Reply:
These are slightly semantically different: what I wrote here is `1..2` or `1..` or `..2` but not `..`. rustc will syntax error if you put `..`, although it will tell you what you did wrong:

```
error: `..` patterns are not allowed here
 --> src/main.rs:3:9
  |
3 |         .. => 4,
  |         ^^
  |
  = note: only allowed in tuple, tuple struct, and slice patterns
```